apps/test/feedbackTests.js

```diff
@@ -225,6 +225,56 @@ describe("throwOnInvalidExampleBlocks", function () {
   });
 });
 
+describe("getUserBlocks_", function () {
+  var studioApp;
+
+  // create our environment
+  beforeEach(function () {
+    testUtils.setupTestBlockly();
+    studioApp = testUtils.getStudioAppSingleton();
+  });
+
+  function validateNumUserBlocks(blockXml, expectedNum) {
+    studioApp.loadBlocks(blockXml);
+
+    // make sure we loaded correctly. text wont match exactly, but make sure if
+    // we had xml, we loaded something
+    var loaded = Blockly.Xml.domToText(Blockly.Xml.blockSpaceToDom(Blockly.mainBlockSpace));
+    assert(loaded, "we didn't correctly load our test blocks");
+
+    var userBlocks = studioApp.feedback_.getUserBlocks_();
+    assert.equal(userBlocks.length, expectedNum);
+  }
+
+  it("usually ignores noneditable blocks", function () {
+    var testBlockXml = [
+      '<xml>',
+      '<block editable="false" type="text_print"></block>',
+      '<block editable="false" type="text"><title name="TEXT">TextContent</title></block>',
+      '<block editable="false" type="math_number"><title name="NUM">10</title></block>',
+      '</xml>'
+    ];
+
+    validateNumUserBlocks(testBlockXml.join(''), 0);
+  });
+
+  it("considers noneditable blocks when Blockly.readOnly === true", function () {
+    var testBlockXml = [
+      '<xml>',
+      '<block editable="false" type="text_print"></block>',
+      '<block editable="false" type="text"><title name="TEXT">TextContent</title></block>',
+      '<block editable="false" type="math_number"><title name="NUM">10</title></block>',
+      '</xml>'
+    ];
+
+    var readOnly = Blockly.readOnly;
+    Blockly.readOnly = true;
```

Comment:
the function itself is [here](https://github.com/code-dot-org/code-dot-org/blob/staging/blockly-core/core/ui/block_space/block_space.js#L208) and the PR that added it is [here](https://github.com/code-dot-org/code-dot-org/pull/6136)